extensions/opencv/src/main/java/ai/djl/opencv/OpenCVImageFactory.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.opencv;
+
+import ai.djl.modality.cv.Image;
+import ai.djl.modality.cv.ImageFactory;
+import ai.djl.modality.cv.util.NDImageUtils;
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.types.DataType;
+import ai.djl.ndarray.types.Shape;
+import ai.djl.util.Utils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Path;
+import nu.pattern.OpenCV;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfByte;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.Imgproc;
+
+/** {@code OpenCVImageFactory} is a high performance implementation of {@link ImageFactory}. */
+public class OpenCVImageFactory extends ImageFactory {
+
+    static {
+        OpenCV.loadLocally();
+        if (System.getProperty("apple.awt.UIElement") == null) {
+            // disables coffee cup image showing up on macOS
+            System.setProperty("apple.awt.UIElement", "true");
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Image fromFile(Path path) throws IOException {
+        Mat img = Imgcodecs.imread(path.toAbsolutePath().toString());
+        if (img.empty()) {
+            throw new IOException("Read image failed: " + path);
+        }
+        return new OpenCVImage(img);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Image fromUrl(URL url) throws IOException {
```

Comment:
Can we make this a default on the ImageFactory?

Reply:
OpenCV dependency is about 85M, I think this should be optional. You can include this extension only if they care about the image reading performance.

Reply:
I mean just the `ImageFactory.fromUrl` as it can be implemented in terms of InputStream

Reply:
done